tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Coverage.java

```diff
@@ -86,8 +86,19 @@ static Coverage getOnlyTheseSources(Coverage coverage, Set<String> sourcesToKeep
     }
     Coverage finalCoverage = new Coverage();
     for (SourceFileCoverage source : coverage.getAllSourceFiles()) {
-      if (sourcesToKeep.contains(source.sourceFileName())) {
-        finalCoverage.add(source);
+      // The manifest lists source files with relative paths, but _cc_coverage.dat has absolute paths.
```

Comment:
We should try to avoid embedding absolute paths into the instrumented binary in the first place. You could add a `coverage` feature to the clang-cl flavor of the Windows toolchain at https://github.com/bazelbuild/rules_cc/blob/main/cc/private/toolchain/windows_cc_toolchain_config.bzl, using unix_cc_toolchain_config.bzl as a reference. You could then add the clang flag `-fcoverage-compilation-dir` to remove the absolute path part.